### PR TITLE
Remove Seq.toList when we already have a list

### PR DESCRIPTION
**Pull request by forki**
This makes use of the unification of our collections

/cc @vasily-kirichenko

### CODE REVIEW

**Comment by smoothdeveloper** [comment]
Might be worth investigating a lint check for those situations in the wild.

File: src/fsharp/NameResolution.fs [hunk 1]
```diff
@@ -3085,7 +3083,6 @@ let rec ResolvePartialLongIdentInModuleOrNamespace (ncenv: NameResolver) nenv is
 
     let ilTyconNames = 
         mty.TypesByAccessNames.Values
-        |> Seq.toList
         |> List.choose (fun (tycon:Tycon) -> if tycon.IsILTycon then Some tycon.DisplayName else None)
         |> Set.ofSeq
```
Comment (vasily-kirichenko) [inline src/fsharp/NameResolution.fs]: `Set.ofList`?

File: src/fsharp/NameResolution.fs [hunk 2]
```diff
@@ -3168,7 +3165,6 @@ let rec ResolvePartialLongIdentPrim (ncenv: NameResolver) (nenv: NameResolutionE
     
        let ilTyconNames =
           nenv.TyconsByAccessNames(fullyQualified).Values
-          |> Seq.toList
           |> List.choose (fun tyconRef -> if tyconRef.IsILTycon then Some tyconRef.DisplayName else None)
           |> Set.ofSeq
```
Comment (vasily-kirichenko) [inline src/fsharp/NameResolution.fs]: same as above

File: src/fsharp/NameResolution.fs [hunk 3]
```diff
@@ -3178,7 +3174,6 @@ let rec ResolvePartialLongIdentPrim (ncenv: NameResolver) (nenv: NameResolutionE
            | FullyQualified -> []
            | OpenQualified ->
                nenv.eUnqualifiedItems.Values
-               |> Seq.toList
                |> List.filter (function Item.UnqualifiedType _ -> false | _ -> true)
                |> List.filter (ItemIsUnseen ad g ncenv.amap m >> not)
```
Comment (vasily-kirichenko) [inline src/fsharp/NameResolution.fs]: merge these two?

File: src/fsharp/NameResolution.fs [hunk 4]
```diff
@@ -3201,7 +3196,6 @@ let rec ResolvePartialLongIdentPrim (ncenv: NameResolver) (nenv: NameResolutionE
 
        let tycons = 
            nenv.TyconsByDemangledNameAndArity(fullyQualified).Values
-           |> Seq.toList
            |> List.filter (fun tcref -> not (tcref.LogicalName.Contains(",")))
            |> List.filter (fun tcref -> not tcref.IsExceptionDecl) 
            |> List.filter (IsTyconUnseen ad g ncenv.amap m >> not)
```
Comment (vasily-kirichenko) [inline src/fsharp/NameResolution.fs]: merge these three?

File: src/fsharp/NameResolution.fs [hunk 5]
```diff
@@ -3266,7 +3259,6 @@ let rec ResolvePartialLongIdentInModuleOrNamespaceForRecordFields (ncenv: NameRe
 
     let ilTyconNames = 
         mty.TypesByAccessNames.Values
-        |> Seq.toList
         |> List.choose (fun (tycon:Tycon) -> if tycon.IsILTycon then Some tycon.DisplayName else None)
         |> Set.ofSeq
```
Comment (vasily-kirichenko) [inline src/fsharp/NameResolution.fs]: same

File: src/fsharp/NameResolution.fs [hunk 6]
```diff
@@ -3333,7 +3325,6 @@ and ResolvePartialLongIdentToClassOrRecdFieldsImpl (ncenv: NameResolver) (nenv:
         // empty plid - return namespaces\modules\record types\accessible fields
        let iltyconNames =
           nenv.TyconsByAccessNames(fullyQualified).Values
-          |> Seq.toList
           |> List.choose (fun tyconRef -> if tyconRef.IsILTycon then Some tyconRef.DisplayName else None)
           |> Set.ofSeq
```
Comment (vasily-kirichenko) [inline src/fsharp/NameResolution.fs]: same

File: src/fsharp/NameResolution.fs [hunk 7]
```diff
@@ -3348,7 +3339,6 @@ and ResolvePartialLongIdentToClassOrRecdFieldsImpl (ncenv: NameResolver) (nenv:
 
        let recdTyCons = 
            nenv.TyconsByDemangledNameAndArity(fullyQualified).Values
-           |> Seq.toList
            |> List.filter (fun tcref -> not (tcref.LogicalName.Contains(",")))
            |> List.filter (fun tcref -> tcref.IsRecordTycon) 
            |> List.filter (IsTyconUnseen ad g ncenv.amap m >> not)
```
Comment (vasily-kirichenko) [inline src/fsharp/NameResolution.fs]: same

File: src/fsharp/PostInferenceChecks.fs [hunk 8]
```diff
@@ -959,11 +959,10 @@ and CheckAttribs cenv env (attribs: Attribs) =
...
         tcrefs 
-        |> Seq.groupBy (fun (tcref,_) -> tcref.Stamp) 
-        |> Seq.map (fun (_,elems) -> List.last (List.ofSeq elems), Seq.length elems) 
-        |> Seq.filter (fun (_,count) -> count > 1) 
-        |> Seq.map fst 
-        |> Seq.toList
+        |> List.groupBy (fun (tcref,_) -> tcref.Stamp) 
+        |> List.map (fun (_,elems) -> List.last elems, List.length elems) 
+        |> List.filter (fun (_,count) -> count > 1) 
+        |> List.map fst
```
Comment (vasily-kirichenko) [inline src/fsharp/PostInferenceChecks.fs]: I'd rollback this change. Long `List.xxx` pipes result with N * (list length) allocations, `Seq.xxx` do not.

File: src/fsharp/NameResolution.fs (hunk 3 above)
Comment (forki) [inline src/fsharp/NameResolution.fs]: that's in the other PR

File: src/fsharp/NameResolution.fs (hunk 4 above)
Comment (forki) [inline src/fsharp/NameResolution.fs]: that's in the other PR

File: src/fsharp/PostInferenceChecks.fs (hunk 8 above)
Comment (forki) [inline src/fsharp/PostInferenceChecks.fs]: done

File: src/fsharp/lib.fs [hunk 9]
```diff
@@ -270,7 +270,7 @@ module List =
         let s = Zset.addList   xs (Zset.empty xOrder) // build set 
         Zset.elements s          // get elements... no repeats 
 
-    let groupBy f (xs:list<'T>) =  xs |> Seq.groupBy f |> Seq.map (map2Of2 Seq.toList) |> Seq.toList
+    let groupBy f (xs:list<'T>) =  xs |> List.groupBy f
```
Comment (dsyme) [inline src/fsharp/lib.fs]: You could just remove this function definition

**Comment by dsyme** [comment]
This is all fine.  I think we can just merge it.

File: src/fsharp/lib.fs (hunk 9 above)
Comment (forki) [inline src/fsharp/lib.fs]: done

**Comment by KevinRansom** [comment]
I like this change too.

**Pull request merged**